### PR TITLE
updated vendoring

### PR DESCRIPTION
**Pull request by simonferquel**
**- What I did**
Updated dependencies so that we can use the new sirupsen import path
**- How I did it**
Had to fork notary
**- How to verify it**
See vendor.conf

### CODE REVIEW

**Comment by aluzzardi** [comment]
Hey @simonferquel:

1) why do we need to update the logrus import path?
2) If we need to do so, why do we need to change the other dependencies?
3) Why do we need to change notary, and if we do need to change it, why not change upstream directly rather than using a fork?

**Comment by vdemeester** [comment]
> why do we need to update the logrus import path?

Because it has been changed upstream (i.e. `moby/moby` for example) and thus, any vendoring update will need to fix thoses.

> If we need to do so, why do we need to change the other dependencies?

Because most of the time the `sirupsen` fix is in _updated_ version of the deps.

> Why do we need to change notary, and if we do need to change it, why not change upstream directly rather than using a fork?

Multiple reason here : 
- cycle vendoring (`libnetwork`, `swarmkit` with `moby/moby` for example — so while it's been merge upstream and update in `moby/moby` we need to vendor the same)
- waiting for PR to get reviewed and merged (`notary`, etc..)
- dependencies that are heavily changed if update (and would need change to our code other than the import) and likely to not getting merged (`rethinkdb` for example)

**Comment by simonferquel** [comment]
@vdemeester can we move along on this PR (don't want to rebase it once again, and only code coverage fails - there is no code addition, only vendoring) ?

File: vendor.conf [hunk 1]
```diff
@@ -1,57 +1,113 @@
...
-github.com/Nvveen/Gotty a8b993ba6abdb0e0c12b0125c603323a71c7790c https://github.com/ijc25/Gotty
-github.com/Sirupsen/logrus v0.11.0
-github.com/agl/ed25519 d2b94fd789ea21d12fac1a4443dd3a3f79cda72c
-github.com/coreos/etcd 824277cb3a577a0e8c829ca9ec557b973fe06d20
-github.com/cpuguy83/go-md2man a65d4d2de4d5f7c74868dfa9b202a3c8be315aaa
-github.com/davecgh/go-spew 346938d642f2ec3594ed81d874461961cd0faa76
-github.com/docker/distribution b38e5838b7b2f2ad48e06ec4b500011976080621
-github.com/docker/docker d58ffa0364c04d03a8f25704d7f0489ee6cd9634
-github.com/docker/docker-credential-helpers v0.5.1
+# docker
```
Comment (dnephin) [inline vendor.conf]: I think we should leave this file sorted. It makes it easier to maintain and avoid conflicts when multiple PRs change this file.  Commented sections are difficult to maintain and will lead to an unsorted mess.  I guess there were a few lines that were not sorted previously. I'll see about adding a validation for this file to keep it sorted.

I checked out this PR and re-sorted the file to look at the vendor changes. I noticed that we have a bunch of new dependencies that I don't think we actually use. Somewhere in the dependency there are some bad packages that are overloaded and forcing us to pull in a bunch of unnecessary extra libraries.  

I'm going to look into this to see if we can remove some of these.

File: vendor.conf [hunk 2]
```diff
@@ -1,57 +1,113 @@
...
 github.com/docker/go-units 9e638d38cf6977a37a8ea0078f3ee75a7cdb2dd1
 github.com/docker/libtrust 9cbd2a1374f46905c68a4eb3694a130610adc62a
-github.com/docker/notary v0.4.2
-github.com/docker/swarmkit 79381d0840be27f8b3f5c667b348a4467d866eeb
-github.com/flynn-archive/go-shlex 3f9db97f856818214da2e1057f8ad84803971cff
-github.com/gogo/protobuf 7efa791bd276fd4db00867cbd982b552627c24cb
+github.com/docker/notary v0.4.2-sirupsen https://github.com/simonferquel/notary.git
+github.com/docker/swarmkit 0554c9bc9a485025e89b8e5c2c1f0d75961906a2
+github.com/moby/buildkit da2b9dc7dab99e824b2b1067ad7d0523e32dd2d9 https://github.com/dmcgowan/buildkit.git
+github.com/containerd/continuity cf279e6ac893682272b4479d4c67fd3abf878b4
```
Comment (dnephin) [inline vendor.conf]: it looks like this is not used, please remove

File: cli/command/image/build_session.go [hunk 3]
```diff
@@ -17,9 +17,9 @@ import (
 	"github.com/docker/cli/cli/command/image/build"
 	cliconfig "github.com/docker/cli/cli/config"
 	"github.com/docker/docker/api/types/versions"
-	"github.com/docker/docker/client/session"
-	"github.com/docker/docker/client/session/filesync"
 	"github.com/docker/docker/pkg/progress"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/filesync"
```
Comment (dnephin) [inline cli/command/image/build_session.go]: Let's keep this for a separate PR. This code is still in `docker/docker/client` in master, so we shouldn't need to do this right now.

File: vendor.conf [hunk 4]
```diff
@@ -1,57 +1,113 @@
...
+github.com/docker/go-events 9461782956ad83b30282bf90e31fa6a70c255ba9
 github.com/docker/go-units 9e638d38cf6977a37a8ea0078f3ee75a7cdb2dd1
 github.com/docker/libtrust 9cbd2a1374f46905c68a4eb3694a130610adc62a
-github.com/docker/notary v0.4.2
-github.com/docker/swarmkit 79381d0840be27f8b3f5c667b348a4467d866eeb
-github.com/flynn-archive/go-shlex 3f9db97f856818214da2e1057f8ad84803971cff
-github.com/gogo/protobuf 7efa791bd276fd4db00867cbd982b552627c24cb
+github.com/docker/notary v0.4.2-sirupsen https://github.com/simonferquel/notary.git
+github.com/docker/swarmkit 0554c9bc9a485025e89b8e5c2c1f0d75961906a2
+github.com/moby/buildkit da2b9dc7dab99e824b2b1067ad7d0523e32dd2d9 https://github.com/dmcgowan/buildkit.git
```
Comment (dnephin) [inline vendor.conf]: this can be removed, we should be using docker/docker/client

File: cli/command/image/build_session.go (hunk 3 above)
Comment (simonferquel) [reply cli/command/image/build_session.go]: It is not there anymore on moby/moby master :(

Comment (dnephin) [reply cli/command/image/build_session.go]: oops, I thought I checked before commenting, but it must have been an old branch. ok to ignore this one

**Comment by simonferquel** [comment]
@dnephin I updated to latest master and was able to remove quite a few dependencies. I went back to a sorted vendor.conf (and sorted it correctly).

**Comment by simonferquel** [comment]
Oh and btw, thanks a lot @dnephin for reducing the dependency surface. It makes this PR far more manageable :)

File: vendor.conf [hunk 5]
```diff
@@ -1,56 +1,59 @@
...
+github.com/docker/go-connections 3ede32e2033de7505e6500d6c868c2b9ed9f169d
 
 # the docker/go package contains a customized version of canonical/json
 # and is used by Notary. The package is periodically rebased on current Go versions.
 github.com/docker/go d30aec9fd63c35133f8f79c3412ad91a3b08be06
-github.com/docker/go-connections 3ede32e2033de7505e6500d6c868c2b9ed9f169d
-github.com/docker/go-events 18b43f1bc85d9cdd42c05a6cd2d444c7a200a894
+github.com/docker/go-events 9461782956ad83b30282bf90e31fa6a70c255ba9
 github.com/docker/go-units 9e638d38cf6977a37a8ea0078f3ee75a7cdb2dd1
+github.com/docker/libnetwork 19ac3ea7f52bb46e0eb10669756cdae0c441a5b1
```
Comment (dnephin) [inline vendor.conf]: I believe this is not used, can be removed.

File: vendor.conf [hunk 6]
```diff
@@ -1,56 +1,59 @@
...
-github.com/docker/notary v0.4.2
-github.com/docker/swarmkit 79381d0840be27f8b3f5c667b348a4467d866eeb
+github.com/docker/notary v0.4.2-sirupsen https://github.com/simonferquel/notary.git
+github.com/docker/swarmkit 0554c9bc9a485025e89b8e5c2c1f0d75961906a2
 github.com/flynn-archive/go-shlex 3f9db97f856818214da2e1057f8ad84803971cff
-github.com/gogo/protobuf 7efa791bd276fd4db00867cbd982b552627c24cb
+github.com/gogo/protobuf v0.4
 github.com/golang/protobuf 7a211bcf3bce0e3f1d74f9894916e6f116ae83b4
-github.com/gotestyourself/gotestyourself v1.0.0
+github.com/google/certificate-transparency 0f6e3d1d1ba4d03fdaab7cd716f36255c2e48341
```
Comment (dnephin) [inline vendor.conf]: not used

File: vendor.conf [hunk 7]
```diff
@@ -1,56 +1,59 @@
...
+github.com/gotestyourself/gotestyourself v1.0.0
 github.com/inconshreveable/mousetrap 76626ae9c91c4f2a10f34cad8ce83ea42c93bb75
 github.com/mattn/go-shellwords v1.0.3
+github.com/Microsoft/go-winio v0.4.4
 github.com/miekg/pkcs11 df8ae6ca730422dba20c768ff38ef7d79077a59f
 github.com/mitchellh/mapstructure f3009df150dadf309fdee4a54ed65c124afad715
-github.com/opencontainers/go-digest a6d0ee40d4207ea02364bd3b9e8e77b9159ba1eb
-github.com/opencontainers/image-spec f03dbe35d449c54915d235f1a3cf8f585a24babe
-github.com/opencontainers/runc 9c2d8d184e5da67c95d601382adf14862e4f2228 https://github.com/docker/runc.git
+github.com/moby/buildkit da2b9dc7dab99e824b2b1067ad7d0523e32dd2d9 https://github.com/dmcgowan/buildkit.git
```
Comment (dnephin) [inline vendor.conf]: can't we get this fix into an official buildkit branch at least? Why is this a fork?

**Comment by andrewhsu** [comment]
@simonferquel @dnephin looks like this has issues when tests run code from moby/moby master branch: https://jenkins.dockerproject.org/job/docker-integration-tests/114/console

**Comment by dnephin** [comment]
I think this hasn't been rebased with master, let me do that now.

**Comment by dnephin** [comment]
done. Can you try again?

**Comment by dnephin** [comment]
Most of the failures seem to be from the panic we fixed a couple days ago.

**Comment by andrewhsu** [comment]
https://jenkins.dockerproject.org/job/docker-integration-tests/115/

**Comment by andrewhsu** [comment]
hmm...seeing new [errors](https://jenkins.dockerproject.org/job/docker-integration-tests/115/consoleFull):
```
[DockerSwarmSuite] FAIL: docker_api_swarm_test.go:349: DockerSwarmSuite.TestAPISwarmRaftQuorum
[DockerSwarmSuite] 
[DockerSwarmSuite] [dfb093aea8ec6] waiting for daemon to start
[DockerSwarmSuite] [dfb093aea8ec6] daemon started
[DockerSwarmSuite] 
[DockerSwarmSuite] [ddc9a034c25d9] waiting for daemon to start
[DockerSwarmSuite] [ddc9a034c25d9] daemon started
[DockerSwarmSuite] 
[DockerSwarmSuite] [df42acc6abc29] waiting for daemon to start
[DockerSwarmSuite] [df42acc6abc29] daemon started
[DockerSwarmSuite] 
[DockerSwarmSuite] [ddc9a034c25d9] exiting daemon
[DockerSwarmSuite] [df42acc6abc29] exiting daemon
[DockerSwarmSuite] docker_api_swarm_test.go:374:
[DockerSwarmSuite]     c.Assert(err.Error(), checker.Contains, "deadline exceeded")
[DockerSwarmSuite] ... obtained string = "Error response from daemon: rpc error: code = Internal desc = transport is closing"
[DockerSwarmSuite] ... substring string = "deadline exceeded"
```

```
[DockerSuite] FAIL: docker_api_containers_test.go:1894: DockerSuite.TestContainersAPICreateMountsCreate
[DockerSuite] 
[DockerSuite] case 0 - config: {volume  /foo false  <nil> <nil> <nil>}
[DockerSuite] case 1 - config: {volume  /foo/ false  <nil> <nil> <nil>}
[DockerSuite] case 2 - config: {volume test1 /foo false  <nil> <nil> <nil>}
[DockerSuite] case 3 - config: {volume test2 /foo true  <nil> <nil> <nil>}
[DockerSuite] case 4 - config: {volume test3 /foo false  <nil> 0xc420539940 <nil>}
[DockerSuite] case 5 - config: {bind /tmp/test-mounts-api-1758465792 /foo false  <nil> <nil> <nil>}
[DockerSuite] docker_api_containers_test.go:2018:
[DockerSuite]     c.Assert(err, checker.NotNil, check.Commentf(out))
[DockerSuite] ... value = nil
[DockerSuite] ... [
[DockerSuite]     {
[DockerSuite]         "Driver": "",
[DockerSuite]         "Labels": null,
[DockerSuite]         "Mountpoint": "",
[DockerSuite]         "Name": "",
[DockerSuite]         "Options": null,
[DockerSuite]         "Scope": ""
[DockerSuite]     }
[DockerSuite] ]
```

and a lot of these:
```
[DockerExternalVolumeSuite] --- FAIL: TestInspect (0.51s)
[DockerExternalVolumeSuite] 	daemon.go:285: [db2c0ab830136] waiting for daemon to start
[DockerExternalVolumeSuite] 	daemon.go:317: [db2c0ab830136] daemon started
[DockerExternalVolumeSuite] 	assertions.go:226:
```

**Comment by dnephin** [comment]
TestInspect is fixed by moby/moby#34682

The other 2 are API tests, so should not be affected by the CLI, probably flakes.

**Comment by dnephin** [comment]
Let's merge this?
